docs/rollup-boost-ha.md

```diff
@@ -0,0 +1,328 @@
+# Table of Contents
+
+- [Context/Scope](#contextscope)
+  - [Goals](#goals)
+  - [Non Goals](#non-goals)
+- [Designs](#designs)
+  - [1:1 Rollup Boost to Builder Deployments](#11-rollup-boost-to-builder-deployments)
+    - [Failure Scenarios](#failure-scenarios)
+    - [Pros/Cons](#pros-and-cons)
+  - [`n` Block Builders](#n-block-builders)
+    - [Failure Scenarios](#failure-scenarios-1)
+    - [Pros/Cons](#pros-and-cons-1)
+  - [Builder Load Balancer](#builder-load-balancer)
+    - [Failure Scenarios](#failure-scenarios-2)
+    - [Pros/Cons](#pros-and-cons-2)
+- [Additional Considerations](#additional-considerations)
+
+# Context/Scope
+
+The current OP Stack sequencer HA design relies on `op-conductor` to manage a cluster of sequencers. Each node runs a local conductor instance, and the cluster forms a Raft quorum to elect a single leader responsible for block production. The conductor continuously monitors the health of the sequencer, electing a new leader when the current leader is unhealthy. The leader notifies it's local sequencer to run in sequencing mode, allowing `op-node` to send FCUs with payload attributes signaling the execution client to build a new payload. All follower instances are run without sequencer mode enabled, ensuring that only one sequencer is producing blocks at a time.
+
+<p align="center">
+  <img src="../assets/op-stack-ha.png" style="width:85%;" />
+</p>
+
+With the introduction of `rollup-boost`, an additional component is introduced that sits in-between `op-node` and `op-geth` that forwards Engine API requests to an external builder.
+
+This design document outlines the architecture, components, and failure strategies required for HA `rollup-boost`. The proposed design prioritizes fault tolerance, liveliness, horizontal scalability, and minimal failover time while maintaining upstream compatibility with `op-conductor`.
+
+## Goals
+- Explore HA designs for `rollup-boost` prioritizing liveliness, fault tolerance and horizontal scalability for external block builders.
+- Maintain compatibility with `op-conductor` and its sequencing assumptions.
+
+## Non Goals
+- Define how Flashblocks are handled, consumed or streamed to the network.
```

Comment:
This is actually a very important part that should be included in this design.

Think about one situation, if say in #2 load balancing solution below, what if get_payload call returns from a different builder that streams to public? This will cause flashblock <-> actual block inconsistencies which we aim to avoid as much as possible.

Reply:
should we update this now? I think we need to take this into consideration when designing HA solutions for flashblocks

Reply:
For clarity, on the design review call last week we aligned on focusing this doc on an HA design that is forward compatible with Flashblocks without defining the full Flashblocks HA specifics here. 

The plan is to fast follow with a separate document that extends the HA design detailing Flashblocks HA behavior including `op-conductor` integration and consistency guarantees. Happy to update and clarify to make this more explicit. Let me know if you have any additional thoughts on this.

Reply:
There is little contradiction in the failover section because of flashblocks.
In https://flashbots.notion.site/Draft_Flashblocks_Eth_OPstack-md-1d66b4a0d876808493c9dcc4428ccad7#1d66b4a0d87680c8b550d356ad113318 it says that we prioritize preconfirmations, so if leader would become partially healthy (builder is down) we will still use block built by builder. In https://github.com/flashbots/rollup-boost/pull/185/files#diff-df3270fd8794e55e7c8f29d84e2605dc61fecaf32696ac4788661c0038e19a6fR242 it says that we will choose block from the EL.